docs/user/implement_algo.md

```diff
@@ -198,11 +198,10 @@ from garage.np import discount_cumsum
 
 class SimpleVPG:
 
-    sampler_cls = LocalSampler
-
-    def __init__(self, env_spec, policy):
+    def __init__(self, env_spec, policy, sampler):
         self.env_spec = env_spec
         self.policy = policy
+        self.sampler = sampler
```

Comment:
Can you make all of the fields here have leading `_`? It demonstrates that they're not part of the policy API.

Reply:
But we need to access algo.sampler in the trainer. Would it be appropriate to make this private? And before this change the `sample_cls` field also did't have leading `_`.

Reply:
It should generally be a read-only property.